src/storage/src/source/reclock/compat.rs

```diff
@@ -103,12 +101,10 @@ pub struct PersistHandle {
     events: LocalBoxStream<'static, ListenEvent<SourceData, (), mz_repr::Timestamp, Diff>>,
     write_handle: WriteHandle<SourceData, (), mz_repr::Timestamp, Diff>,
     snapshot_produced: bool,
-    upper: Antichain<mz_repr::Timestamp>,
     as_of: Antichain<mz_repr::Timestamp>,
-    pending_batch: Vec<((PartitionId, MzOffset), mz_repr::Timestamp, Diff)>,
-    native_source_upper: MutableAntichain<Partitioned<PartitionId, MzOffset>>,
-    compat_source_upper: MutableOffsetAntichain,
-    minimum_produced: bool,
+    pending_batch: Vec<(Partitioned<PartitionId, MzOffset>, mz_repr::Timestamp, Diff)>,
+    synthesize_partition_minimums: bool,
```

Comment:
This field seems redundant as it's inferred from `partitioned_source`

Reply:
This is part of the remapping state machine, i.e. it is set to false after seeing the first non-minimum data for non-partitioned sources.